packages/graph/hash_graph/lib/graph/src/types/mod.rs

```diff
@@ -0,0 +1,21 @@
+//! Descriptions, and implementation logic, of the Types of elements of the graph
+
+/// A placeholder enum denoting the possible types
+pub enum GraphElementType {
```

Comment:
Not sure if this enum is a good idea at all, was just experimenting

Reply:
I think making it a trait may be better. I'll have a deeper look tomorrow and will review it properly afterward.